components/experimental/src/dimension/units/format.rs

```diff
@@ -71,6 +71,23 @@ fn test_basic() {
     formatted_unit.write_to(&mut sink).unwrap();
     assert_eq!(sink.as_str(), "12,345.67 m");
 
+    let locale = locale!("en-US").into();
+    let century = "century";
+    let fmt = UnitsFormatter::try_new(
+        &locale,
+        century,
+        UnitsFormatterOptions {
+            width: Width::Long,
+            ..Default::default()
+        },
+    )
+    .unwrap();
+    let value = "12345.67".parse().unwrap();
+    let formatted_unit = fmt.format_fixed_decimal(&value, meter);
```

Comment:
why does this use `meter`? it doesn't seem to fail

Reply:
Good catch, that is because this `Id` will be used only if there is no data associated with `MeasureUnit`. we use the  id.

I will file a bug to have this id associated with the formatter.

Reply:
https://github.com/unicode-org/icu4x/issues/5227